### ISSUE TITLE
Optimizer crash in mesh code

### ISSUE THREAD

**Issue by akenmorris** [SCIInstitute/ShapeWorks#925]
Here is the stacktrace from a user:

```
#0 0x000000000067f977 in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#1 0x000000000067fb88 in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#2 0x000000000067fb88 in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#3 0x000000000067fb88 in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#4 0x000000000067fbbb in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#5 0x000000000067faad in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#6 0x000000000067fb88 in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#7 0x000000000067fb88 in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#8 0x000000000067fb88 in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#9 0x000000000067faad in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#10 0x000000000067faad in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#11 0x000000000067faad in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#12 0x000000000067fb88 in trimesh::KDtree::Node::find_k_closest_to_pt(trimesh::KDtree::Node::Traversal_Info&) const ()
#13 0x0000000000680755 in trimesh::KDtree::find_k_closest_to_pt(std::vector<float const*, std::allocator<float const*> >&, int, float const*, float, trimesh::KDtree::CompatFunc const*, float) const ()
#14 0x0000000000664a64 in shapeworks::TriMeshWrapper::GetKNearestVertices(trimesh::Vec<3ul, float>, int) const ()
#15 0x0000000000669cca in shapeworks::TriMeshWrapper::GetTriangleForPoint(trimesh::Vec<3ul, float>) const ()
#16 0x000000000066a148 in shapeworks::TriMeshWrapper::SampleNormalAtPoint(itk::Point<double, 3u>) const ()
#17 0x000000000066c969 in itk::MeshDomain::SampleNormalAtPoint(itk::Point<double, 3u> const&) const ()
#18 0x000000000063d7fa in itk::ParticleSurfaceNeighborhood<itk::Image<float, 3u> >::FindNeighborhoodPoints(itk::Point<double, 3u> const&, std::vector<double, std::allocator<double> >&, double) const ()
#19 0x00000000005f6f12 in itk::ParticleCurvatureEntropyGradientFunction<float, 3u>::BeforeEvaluate(unsigned int, unsigned int, itk::ParticleSystem<3u> const*) ()
#20 0x00000000005e2ea8 in itk::ParticleDualVectorFunction<3u>::BeforeEvaluate(unsigned int, unsigned int, itk::ParticleSystem<3u> const*) ()
#21 0x00000000005c16bb in itk::ParticleGradientDescentPositionOptimizer<float, 3u>::StartAdaptiveGaussSeidelOptimization() [clone ._omp_fn.0] ()
#22 0x00007f14ba94248e in ?? () from /usr/lib64/libgomp.so.1
#23 0x00007f14baf4a4f9 in start_thread () from /lib64/libpthread.so.0
#24 0x00007f14ba44efbf in clone () from /lib64/libc.so.6
```

We have been unable to reproduce it so far.

**Comment by sheryjoe** [comment]
@akenmorris @medakk If this is still an issue, it should be resolved before the 6 release.

**Comment by sheryjoe** [comment]
Related to #926

**Comment by akenmorris** [comment]
Fixed in #1065
